src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp

```diff
@@ -681,13 +681,13 @@ void InterpreterMacroAssembler::unlock_if_synchronized_method(TosState state,
     // Check if synchronized method or unlocking prevented by
     // JavaThread::do_not_unlock_if_synchronized flag.
     lbz(Rdo_not_unlock_flag, in_bytes(JavaThread::do_not_unlock_if_synchronized_offset()), R16_thread);
-    lwz(Raccess_flags, in_bytes(Method::access_flags_offset()), R19_method);
+    lhz(Raccess_flags, in_bytes(Method::access_flags_offset()), R19_method);
     li(R0, 0);
     stb(R0, in_bytes(JavaThread::do_not_unlock_if_synchronized_offset()), R16_thread); // reset flag
 
     push(state);
 
-    // Skip if we don't have to unlock.
+    // Skip if we don't have to unlock. (???is this right???)
     rldicl_(R0, Raccess_flags, 64-JVM_ACC_SYNCHRONIZED_BIT, 63); // Extract bit and compare to 0.
```

Comment:
Using `testbitdi` might make it more readable to non-experts.  It took me a while reading aix docs to realize that this platform numbers LSB as 63 and MSB/sign as 0.

Reply:
yes I like testbitdi better.  I found a sample in the templateInterpreterGenerator code.

Reply:
Right, Bit 0 refers to the MSB on IBM platforms (not only AIX). Using `testbitdi` is also fine.